src/plugins/kibana_utils/public/history/ensure_default_index_pattern.tsx

```diff
@@ -42,24 +42,19 @@ export async function ensureDefaultIndexPattern(
   data: DataPublicPluginStart,
   history: History
 ) {
-  const patterns = await data.indexPatterns.getIds();
-  let defaultId = core.uiSettings.get('defaultIndex');
-  let defined = !!defaultId;
+  const patterns = await data.indexPatterns.getIds(true);
+  const defaultId = core.uiSettings.get('defaultIndex');
+  const defined = !!defaultId;
   const exists = contains(patterns, defaultId);
 
-  if (defined && !exists) {
-    core.uiSettings.remove('defaultIndex');
-    defaultId = defined = false;
-  }
-
-  if (defined) {
-    return;
+  if (defined && exists) {
+    return defaultId;
   }
 
   // If there is any index pattern created, set the first as default
   if (patterns.length >= 1) {
-    defaultId = patterns[0];
-    core.uiSettings.set('defaultIndex', defaultId);
+    return patterns[0];
+    // core.uiSettings.set('defaultIndex', defaultId);
```

Comment:
Delete comment

Reply:
@lizozom was removed in the recent cleanup, thx